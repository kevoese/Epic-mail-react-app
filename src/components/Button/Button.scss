.btn {
  font-family: cambria;
  font-size: 0.75em;
  white-space: nowrap;
  padding: 1em 3em;
  cursor: pointer;
  display: block;
  color: white;
  position: relative;
  background-color: var(--myColor);
  border: 1px solid var(--myColor);
  border-radius: 3em;

&:hover {
  box-shadow: 4px 4px 7px -3px rgba(0, 0, 0, 0.5);
}
}