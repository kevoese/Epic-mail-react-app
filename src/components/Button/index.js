/* eslint-disable react/button-has-type */
import React from 'react';
import './Button.scss';

const Button = ({
  classes, children, isSubmitting, onClick, ...rest
}) => (
  <button disabled={isSubmitting} {...rest} onClick={onClick} className={`btn ${classes || ''}`}>
    {isSubmitting ? 'Loading...' : children}
  </button>
);

export default Button;
