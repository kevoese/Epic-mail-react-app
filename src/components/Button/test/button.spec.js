/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import 'jsdom-global/register';
import Button from '@components/Button';

describe('Button component', () => {
  it('should render the button without loading when submitting is false', () => {
    const handleClick = jest.fn();
    const component = shallow(
      <Button isSubmitting={false} onClick={handleClick} classes="">
        test
      </Button>
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the button with loading when submitting is true', () => {
    const handleClick = jest.fn();
    const component = shallow(
      <Button isSubmitting={false} onClick={handleClick} classes="">
        test
      </Button>
    );
    expect(component).toMatchSnapshot();
  });
});
