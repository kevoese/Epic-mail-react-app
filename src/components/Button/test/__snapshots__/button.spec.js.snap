// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render the button with loading when submitting is true 1`] = `
<button
  className="btn "
  disabled={false}
  onClick={[MockFunction]}
>
  test
</button>
`;

exports[`Button component should render the button without loading when submitting is false 1`] = `
<button
  className="btn "
  disabled={false}
  onClick={[MockFunction]}
>
  test
</button>
`;
